Addition to `match` section (#87)
darklang/docs: docs/functional-aspects.md

## Match

> The match expression is used to destructure complex types. It can be used similarly to switch statements in other languages. It supports the Option type (Just and Nothing) and Result type (Ok and Error).

Reviewer: ```suggestion
The match expression is used to destructure complex types, such as the `Option` type (`Just` and `Nothing`) and the `Result` type (`Ok` and `Error`). It can also be used similarly to switch statements in other languages. 
```

By default, functions that return Option or Result go to the [Error Rail](https://darklang.github.io/docs/unique-aspects#functions-that-use-error-rail). Once you remove the function from the Error Rail, use match for destructuring.

Here’s an example of using a match expression with a dictionary lookup.

![assets/functional/image16.png](assets/functional/image16.png)

> Using any variable name as one of the branches will catch all matches. The variable can be used on the right side of the branch.

Reviewer: ```suggestion
Using a variable name as one of the branches will catch all matches. The variable can be used on the right side of the branch.
```
